### DISCUSSION TITLE
Scoppy is now also a Logic Analyzer

### DISCUSSION THREAD

**Discussion by fhdm-dev** [fhdm-dev/scoppy#26] [General]
New versions of the Scoppy firmware and app have been released which add some rudimentary logic analyzer functionality.
The maximum sample rate is 20MS/s per channel and there are 8 channels (2 are available in the free version of the app).

> Please note that the pin assignments for the test signal and voltage ranges have changed. The test signal is now on GPIO22 and the voltage range inputs are now GPIO2 to GPIO5.

Once you've upgraded the app you'll need the [latest version of the firmware](https://github.com/fhdm-dev/scoppy/wiki/Installation-&-Getting-Started).

**Comment by santlaha** [comment]
Wow. I was eagerly waiting for this. What is the max speed?

**Reply by fhdm-dev** [reply]
> Wow. I was eagerly waiting for this. What is the max speed?

It seems to work OK for 1MHz signals. Higher than that the triggering starts to get a bit wonky.

**Comment by santlaha** [comment]
that's enough for now.


On Thu, Aug 5, 2021 at 12:28 PM fhdm-dev ***@***.***> wrote:

> Wow. I was eagerly waiting for this. What is the max speed?
>
> It seems to work OK for 1MHz signals. Higher than that the triggering
> starts to get a bit wonky.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/fhdm-dev/scoppy/discussions/26#discussioncomment-1133258>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUQS6G6VVVD5QNKND3J5YBLT3IZCVANCNFSM5BSXDTOA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Santanu Laha

**Comment by santlaha** [comment]
I think a simple voltage divider should work fine for 5 v logic signals.

On Thu, Aug 5, 2021 at 12:28 PM fhdm-dev ***@***.***> wrote:

> Wow. I was eagerly waiting for this. What is the max speed?
>
> It seems to work OK for 1MHz signals. Higher than that the triggering
> starts to get a bit wonky.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/fhdm-dev/scoppy/discussions/26#discussioncomment-1133258>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUQS6G6VVVD5QNKND3J5YBLT3IZCVANCNFSM5BSXDTOA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Santanu Laha

**Reply by fhdm-dev** [reply]
> I think a simple voltage divider should work fine for 5 v logic signals.

It would have if I hadn't forgotten to disable the pull-downs on the input gpios :grin: 

This is now fixed in [version 5](https://scoppy.fhdm.xyz/downloads/scoppy-pico-v5.uf2) of the firmware.

![Uno I2C](https://github.com/fhdm-dev/scoppy/raw/main/images/20210808-uno-i2c-1.jpg)
`UNO controlling an OLED display using I2C - SCL connected to GPIO6 on the Pico, SDA on GPIO9`

**Comment by rog77** [comment]
Would there be any way to add more channels for the logic analyser? I would like to poke at 7400 series logic circuits, so as many as possible would be preferable, but for my purposes they wouldn't need to be very fast. Thanks either way.

**Reply by fhdm-dev** [reply]
Hi. This is definitely possible but would require significant changes to both the firmware and app so it's probably not something that will happen anytime soon (I'm concentrating more on the oscilloscope side of things).
Have you tried [sigrok-pico](https://github.com/pico-coder/sigrok-pico)?

**Reply by rog77** [reply]
Thanks for the response, I will certainly give that a go.

As an aside, since I expect you know about these things, I read that the Pico ADC does per bit round Robin SAR, would that mean that in the can of taking two analog samples they are stored and held at the same moment in time? I am also interested in sampling I/Q for SDR.

Once again thank you for the response. I did pay for the upgraded version of scoppy, and I think it it is very cool :-)

**Reply by fhdm-dev** [reply]
Not at exactly the same moment. The second channel won't be sampled until the first is complete.
Thank's for upgrading. Much appreciated :grinning:
